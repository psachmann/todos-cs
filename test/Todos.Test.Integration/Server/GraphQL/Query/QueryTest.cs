namespace Todos.Test.Integration.Server.GraphQL;

public class QueryTest : IClassFixture<TodosServerFixture>
{
    private readonly TodosServerFixture _fixture;

    public QueryTest(TodosServerFixture fixture)
    {
        _fixture = fixture;
    }

    [Fact]
    public async Task GetTodoItem_ShouldReturnTodoItem()
    {
        var result = await _fixture.ExecuteQueryAsync((request) => request.SetQuery(
            """
            {
                todoItem {
                    id,
                    title,
                    isDone,
                }
            }
            """));

        result.Should().NotBeNull();
        result.MatchSnapshot();
    }

    [Fact]
    public async Task GetTodoList_ShouldReturnTodoList()
    {
        var id = Guid.NewGuid();
        var result = await _fixture.ExecuteQueryAsync((request) => request.SetQuery(
            $$""""
            {
                todoList(query: { {{id}} }) {
                    id,
                    title,
                    isDone,
                }
            }
            """"));

        result.Should().NotBeNull();
        result.MatchSnapshot();
    }
}
