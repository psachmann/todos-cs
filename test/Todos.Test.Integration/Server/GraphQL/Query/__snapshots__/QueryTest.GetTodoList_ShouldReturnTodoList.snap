{
  "errors": [
    {
      "message": "Expected a \u0060Name\u0060-token, but found a \u0060Float\u0060-token.",
      "locations": [
        {
          "line": 2,
          "column": 23
        }
      ],
      "extensions": {
        "code": "HC0014"
      }
    }
  ]
}
